src/elf_sections.rs

```diff
@@ -53,22 +57,27 @@ impl StringTable {
 
 #[derive(Clone)]
 pub struct ElfSectionIter {
-    current_section: &'static ElfSection,
+    current_section: &'static u8,
     remaining_sections: u32,
     entry_size: u32,
 }
 
 impl Iterator for ElfSectionIter {
     type Item = &'static ElfSection;
+
     fn next(&mut self) -> Option<&'static ElfSection> {
         if self.remaining_sections == 0 {
             None
         } else {
-            let section = self.current_section;
+            let section: &'static ElfSection = match self.entry_size {
+                40 => unsafe { &*(self.current_section as *const _ as *const ElfSection32) },
+                64 => unsafe { &*(self.current_section as *const _ as *const ElfSection64) },
```

Comment:
I would prefer `core::mem::size_of<ElfSection32|64()` to comparing with `40` and `60` if they are equivilant.